wireapp/kalium
# fix: handle member join when being added to group (AR-1673)

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When self user is added to an ALREADY EXISTING conversation, it doesn't show-up in the conversation list until a full sync is performed.

### Causes

✅ When the conversation is new, we receive `Event.Conversation.NewConversation`. Which we are properly handling. 

❌ When the conversation ALREADY EXISTS, and we are added to it, we receive `Event.Conversation.MemberJoin`. Currently we are only adding members, ignoring the possibility that this conversation is completely unknown. SQL Constraints will fail, because the conversation is unknown.

### Solutions

- When receiving `Event.Conversation.MemberJoin`, try to fetch the details of this conversation.
- Even if it failed (for whatever reason), try to add members to the conversation to avoid missing the event.

### Testing

#### Test Coverage


#### How to Test

1. As User A, create a group conversation without members
2. As User B, observe the conversations list.
3. As User A, add User B to the group conversation
4. As User B, you should be able to see the new conversation on the very top.


### Notes

- `ConversationRepository$persistMembers` was refactored to accept business logic models instead of Entities.
- `ConversationRepository$persistMember` (the singular version) was removed as it wasn't being used.

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
